### ISSUE TITLE
Lint against PEP 810 lazy imports that would be immediately resolved

### ISSUE THREAD

**Issue by JelleZijlstra** [astral-sh/ruff#25014]
### Summary

With the new PEP-810 lazy imports in Python 3.15, it's possible to write code like this:

```python
lazy import foo

class Bar(foo.Foo): ...
```

where the "lazy" import gets resolved immediately at module load time. That isn't what the user intended, so linters should warn against it.

The rule should check for names created through a lazy import and used either in the module global scope (and not within an except block or conditional block) or in a class body that is immediately within the global scope. More ambitiously, we could also check function bodies that we detect are called directly from the module scope (e.g. decorators, dataclass annotations). Doing this in full generality is obviously intractable, but some heuristics could go a long way.

I'm not sure the rule should have an autofix. Removing "lazy" is an obvious solution, but it may not be what the user wants: if they wrote `lazy` they probably intended it to be lazy, and we should give them a chance to fix their code.

**Comment by charliermarsh** [comment]
Up here: https://github.com/astral-sh/ruff/pull/25016
